src/raster/r.hydro.flatten/r.hydro.flatten.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python
+#
+#########################################################################
+#
+# MODULE:     r.hydro.flatten
+#
+# AUTHOR(S):  Anna Petrasova <kratochanna gmail com>
+#
+# PURPOSE:    Derive elevation of water bodies for hydro-flattening
+#
+# COPYRIGHT:  (C) 2023 by Anna Petrasova, and the GRASS Development Team
+#
+#             This program is free software under the GNU General Public
+#             License (>=v2). Read the COPYING file that comes with GRASS
+#             for details.
+#
+#########################################################################
+
+# %module
+# % description: Derive elevation of water bodies for hydro-flattening
+# % keyword: raster
+# % keyword: elevation
+# % keyword: hydrology
+# % keyword: lidar
+# % keyword: LIDAR
+# %end
+# %option G_OPT_R_INPUT
+# % key: input
+# % description: Raster map of binned lidar point elevation
+# %end
+# %option G_OPT_R_OUTPUT
+# % key: water_elevation
+# % description: Raster map of derived water elevation
+# %end
+# %option G_OPT_R_OUTPUT
+# % key: water_elevation_stddev
+# % description: Raster map of derived water elevation standard deviation
+# %end
+# %option
+# % key: percentile
+# % type: double
+# % required: yes
+# % description: Percentile of elevation to determine water level
+# % answer: 5
+# %end
+# %option
+# % key: minsize
+# % type: integer
+# % required: no
+# % description: Minimum size of areas in map units
+# %end
+# %flag
+# % key: k
+# % description: Keep intermediate results
+# %end
+
+import os
+import sys
+import atexit
+from math import sqrt
+
+import grass.script as gs
+
+RAST_REMOVE = []
+
+
+def cleanup():
+    if RAST_REMOVE:
+        gs.run_command("g.remove", flags="f", type="raster", name=RAST_REMOVE)
+
+
+def get_tmp_name(basename):
+    name = gs.append_node_pid(basename)
+    RAST_REMOVE.append(name)
+    return name
+
+
+def get_intermediate_name(basename):
+    return f"intermediate_{basename}"
+
+
+def main():
+    options, flags = gs.parser()
+    keep = flags["k"]
+    if keep:
+        get_name = get_intermediate_name
```

Comment:
Passing the main output name here through lambda would be a little nicer and safer.

Reply:
I used lambda and was getting some flake8 issues.

Reply:
Then "all" you need is a nested function, just write `def get_name` there into the if. That seems to be the preferred solution. I guess more lines, but possibly more readable in general.